# Activity 3: Practicing with Pandas

For this activity, we will assume the role of a data scientist in the search for and evaluation of public datasets. In turn, pay close attention to the dataset you choose for counter narratives or nonapparent political values, gaps or underrepresented perspectives.

Please make sure you open this file in Jupyter Notebook or Google Colab -- you will have to either open Jupyter Notebook first or upload the file to Google Colab before being able to edit the file.

Please make sure you **print your output.** You may need to use ```print``` for your output to be displayed.

-----
## Submission Instructions
1. Upload the completed activity to the "Activities" folder of your GitHub Portfolio;
2. Ensure your submission consists of two files: <br/>(a) one single-page reflection in markdown (e.g. `a3_reflection.md`);  <br/>(b) one Jupyter notebook with completed responses to each activity step (e.g. `a3_pandas.ipynb`).

## Data biography: Trans-Atlantic Slave Trade dataset

To complete the assignment, please make sure you download the dataset from where it hosted on our GitHub: [Trans-Atlantic-Slave-Trade_Americas.csv](https://github.com/zmuhls/ccny-data-science/blob/main/assets/datasets/Trans-Atlantic-Slave-Trade_Americas.csv)

## Data Biography (1-2 Paragraphs)

Before you begin your exploration, you will provide a short data biography (1 - 2 paragraphs) on the dataset that addresses **where the data came from, who collected it, and the original intention(s) for its collection.**

You may find the following sources useful for your research:
* [Slave Voyages' Trans-Atlantic Slave Trade methodology](https://www.slavevoyages.org/voyage/about#methodology/introduction/0/en/)
    * You may find the sections: Introduction, Coverage of the Slave Trade, and Nature of Sources particularly useful.
* [Slave Voyages' About](https://www.slavevoyages.org/about/about#)
* [Jamelle Bouie's We Still Can’t See American Slavery for What It Was](https://csc10800.github.io/assets/pdf/Bouie_we_still_cant_see_american_slavery_for_what_it_is.pdf)

**Enclose Data Biography below (1-2 Paragraph)**:
After reading the article "We Still Can't See Slavery For What It Was" by Jamelle Bouie, the article gave me clear insight on how exactly data was being collected during slavery and why. Well, first the reason why is because this data is of course a part of history; furthermore, has to exactly let historians, teachers, observers or readers know exactly the population of slaves being human trafficked and shipped from the western coast of Africa to the Western Hemisphere.
According to this article, I found out that 12.5 million people were human trafficked and were being brought to the western hemisphere. How was this found out? Well, that is because people such as students and historians are people who found this information. They kept track of the amount slave voyages



> Add blockquote




-----

## Download and explore dataset

As we have been practicing in class, the questions that follow will require you to consider what output you will need before running the appropriate python codes.

To ensure your code runs, please remember to load ```pandas``` and the ```Trans-Atlantic-Slave-Trade_Americas.csv``` file into the jupyter notebook environment.

If you are running this on Google Colab, please make sure you connect your google drive with your Colab notebook first before attempting the exercise:
```
from google.colab import drive
drive.mount('/content/drive')
```

In [ ]:
import pandas as pd
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


### 2. How many rows of data does the dataset contain?</br>
In this first step, you’ll load the dataset and examine its structure. Loading data is foundational in data science and digital humanities workflows. You’ll use the `pandas` library to read a CSV file and display its first few rows.</br>
* **Hint**: See Melanie Walsh's chapter on Pandas Basics if you're stuck: [Pandas Basics – Part 1](https://melaniewalsh.github.io/Intro-Cultural-Analytics/03-Data-Analysis/01-Pandas-Basics-Part1.html)).



In [ ]:
slave_voyages_df = pd.read_csv('/content/Trans-Atlantic-Slave-Trade_Americas.csv', delimiter=",", encoding='utf-8')

In [ ]:
slave_voyages_df

year_of_arrival               flag          place_of_purchase  \
0                 1520                NaN          Portuguese Guinea   
1                 1525  Portugal / Brazil                   Sao Tome   
2                 1526    Spain / Uruguay         Cape Verde Islands   
3                 1526    Spain / Uruguay         Cape Verde Islands   
4                 1526                NaN         Cape Verde Islands   
...                ...                ...                        ...   
20774             1864    Spain / Uruguay  Africa., port unspecified   
20775             1865    Spain / Uruguay  Africa., port unspecified   
20776             1865                NaN  Africa., port unspecified   
20777             1865                NaN                Congo River   
20778             1866                NaN  Africa., port unspecified   

                     place_of_landing  percent_women  percent_children  \
0                            San Juan            NaN               NaN   
1             Hispaniola, unspecified            NaN               NaN   
2              Cuba, port unspecified            NaN               NaN   
3              Cuba, port unspecified            NaN               NaN   
4      Caribbean (colony unspecified)            NaN               NaN   
...                               ...            ...               ...   
20774          Cuba, port unspecified            NaN               NaN   
20775                   Isla de Pinas            NaN               NaN   
20776                          Mariel            NaN               NaN   
20777          Cuba, port unspecified            NaN               NaN   
20778          Cuba, port unspecified            NaN               NaN   

       percent_men  total_embarked  total_disembarked    resistance_label  \
0              NaN           324.0              259.0                 NaN   
1              NaN           359.0              287.0                 NaN   
2              NaN           359.0              287.0                 NaN   
3              NaN           359.0              287.0                 NaN   
4              NaN           359.0              287.0                 NaN   
...            ...             ...                ...                 ...   
20774          NaN           488.0              465.0                 NaN   
20775          NaN           152.0              145.0  Slave insurrection   
20776          NaN           780.0              650.0                 NaN   
20777          NaN          1265.0             1004.0                 NaN   
20778          NaN           851.0              700.0                 NaN   

             vessel_name       captain's_name  voyage_id  \
0                                         NaN      42987   
1      S Maria de Bogoña       Monteiro, Pero      46473   
2                         Carega, Esteban (?)      11297   
3                         Carega, Esteban (?)      11298   
4                S Anton        Leon, Juan de      42631   
...                  ...                  ...        ...   
20774             Polaca                  NaN      46554   
20775               Gato                  NaN       4394   
20776                                     NaN       4395   
20777            Cicerón             Mesquita       5052   
20778                                     NaN       4998   

                                                 sources  
0      [u'AGI,Patronato 175, r.9<><p><em>AG!</em> (Se...  
1      [u'ANTT,CC,Parte II, maco 131, doc 54<><i>Inst...  
2      [u'Pike,60-1,172<>Pike, Ruth, <i>Enterprise</i...  
3      [u'Pike,60-1,172<>Pike, Ruth, <i>Enterprise</i...  
4      [u'Chaunus, 3: 162-63<><p>Chaunus, <em>xxxxxx<...  
...                                                  ...  
20774  [u'AHNM, Ultramar, Leg. 3551, 6<><i>Archivo Hi...  
20775  [u'IUP,ST,50/B/137<>Great Britain, <i>Irish Un...  
20776  [u'IUP,ST,50/B/144<>Great Britain, <i>Irish Un...  
20777  [u'IUP,ST,

i mul0 mj**Double click to edit cell**     
`Please type your answer here`
</br>
</br>
</br>

---


### 3. Are the data types for each column appropriate? Please explain how they are/not approrpiate for your analysis.

**Double click to edit cell**     
`Please type your answer here`
In the beginning of the chart you can see that there are two results for Spain/ Urguguay. They both have the same results and the same year; however, the error that appears to be there is that they both have different voyage Id's. The first voyage Id being 11297 and the other being 11298. the reason why this isn't appropiate is due to the fact that it just seems funny how for in Spain during 1526 both information that was collected were the same in terms of the total amount of people embarked and disembarked also, it shows the same captain for both which is Estaban Carega ; however, they have different voyage Id's. Another reason is that for each column you can see that the data that was collected does not tell whether not if the person is a child, a man, or a woman.

---


### 4. What is the overall average proportion of ```percent_women```, ```percent_children```, ```percent_men```?

**Double click to edit cell**     
The overall average proprtion of women, children, and men can't be defined; therefore, it would be zero.


### 5. How many of the column values for ```percent_women```, ```percent_children```, ```percent_men``` are left blank? Suggest 1 reason why majority of the values in these columns are blank.

In [ ]:
slave_voyages_df{'percentage_women','percentage_men','percentage_children'}.columms=10
print(slave_voyages_df{'percentage_women','percentage_men','percentage_children'}.columns)
10

In [ ]:
10

10

**Double click to edit cell**     
There are 10 column values that are left blank for woman, man, and children. one of the reason why it could be listed as blank is due to the fact that they were jst described as propety and not people. Also, they weren'r born in hospitals after giing birth to a baby, the baby becomes a slave; furthermore, doesn't visit the hospital nor the man or woman.

---


### 6. Display all duplicated rows and remove the duplicates. Check that the duplicates were successfully removed. Please also suggest a reason why we would remove duplicates for our analysis.

**Level-up**: How many duplicated rows do we have? Recall the methods we have used to help us count things.

In [ ]:
N/A

0

**Double click to edit cell**     
`Please type your answer here`
</br>
</br>
</br>


---


### 7. Please identify the **top 5 most common ports of arrival**.

**Hint:** Check the columns ```place_of_landing```.

In [ ]:
#Your code here

**Double click to edit cell**     
Cuba, Mariel, Hispaniola, San Juan, Isla de pinas

---


### 8. Please plot the **top 5 enslavers/captors** from this dataset. Please choose the appropriate visualization (e.g. pie chart, bar chart) to display your finding.

In [ ]:
slave_voyages_df{top_five_enslavers}='Mesquiwta','Voyage Id 4395','Voyage Id 4998','Voyage Id 46554','Monteiro, Pero 4647'
print(slave_voyages_df{top_five_enslavers})
'Mesquiwta','Voyage Id 4395','Voyage Id 4998','Voyage Id 46554','Monteiro, Pero 4647'


-----

### 9. Having briefly explored the dataset, what further questions have emerged as you explored this dataset?    

Please share your thoughts in a few sentences.

**Double click to edit cell**     
is there any other possible way to know the exact percentage of men, women, and chilren during slavery although theyh were all listed as property? is the amount of people embarked and disembarked during slavery actually true meaning are these results actually factual. How do we know that there weren't any more mistakes knowing that we saw mistakes in the previious chart? Is this something that we can rely on or no?


-----
-----


## Side quest challenge

The side quest challenge is for extra credit. You can still get full credit for this activity even if you do not complete this challenge.

### 1. Which enslaver/captor had the highest difference between the total number of people who embarked and disembarked?

**Hint:** You may need to [add a column](https://github.com/GCDigitalFellows/intro-pandas-dri-2022/blob/main/README.md#8-rename-select-drop-and-add-new-columns) to calculate the difference, use the [```.groupby```, ```.count()``` and ```.sort_values``` methods](https://github.com/GCDigitalFellows/intro-pandas-dri-2022/blob/main/README.md#9-sort-columns-groupby-columns--count-values) for this challenge.

**Double click to edit cell**     
`Please type your answer here`
</br>
</br>
</br>

---
